apps/client/src/module/appsetting/app-setting-page.tsx

```diff
@@ -0,0 +1,93 @@
+"use client";
+import { useTranslations } from "next-intl";
+import { Select } from "../common/select";
+import { useForm } from "react-hook-form";
+import { useRouter } from "../../configs/navigation";
+import { useLanguageSwitcher } from "./language-switcher";
+import { useThemeSwitcher } from "./theme-switcher";
+import { useTheme } from "../system/theme-switcher";
+
+type FormValues = {
+  language: string;
+  theme: string;
+};
+
+export function AppSettingPage() {
+  const t = useTranslations();
+  const router = useRouter();
+  const {
+    isLocale,
+    switchLanguageValue,
+    currentLocale,
+    localesSwitchRules,
+    loading,
+  } = useLanguageSwitcher();
+  const {
+    currentTheme,
+    setCurrentTheme,
+    switchThemeToOption,
+    switchOptionToTheme,
+    isTheme,
+  } = useThemeSwitcher();
+  const { setTheme } = useTheme();
+
+  const options = {
+    language: ["English", "中文"],
+    theme: [t("Shared.darkTheme"), t("Shared.lightTheme")],
+  };
+
+  const { control } = useForm<FormValues>({
+    defaultValues: {
+      language: "",
+      theme: "",
+    },
+  });
+
+  const changeLanguageHandler = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const selectedLanguage = e.target.value as keyof typeof localesSwitchRules;
+    const destinationLocale = switchLanguageValue(selectedLanguage);
+    if (isLocale(destinationLocale)) {
+      router.replace("/appsetting", { locale: destinationLocale });
+    }
+  };
+
+  const changeThemeHandler = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const selectedTheme = e.target.value;
+    const theme = switchOptionToTheme(selectedTheme);
+    if (isTheme(theme)) {
+      setTheme(theme);
+      setCurrentTheme(theme);
+    }
+  };
+
+  return (
+    <div className="dark:bg-primary-900 bg-primary-50 shadow-settingPage dark:shadow-settingPage flex flex-col rounded-lg p-5">
+      <div className="space-y-1">
+        <h1 className="text-base font-bold">{t("AppSettingPage.title")}</h1>
+        <h5 className="text-primary-700 dark:text-primary-100 text-sm opacity-70">
+          {t("AppSettingPage.subtitle")}
+        </h5>
+      </div>
+      {!loading && (
+        <form className="my-[30px] flex flex-col gap-y-5">
+          <Select
```

Comment:
again, it doesn't seems like you need a controlled select here.

Reply:
yeah， already delete react form hook